File: cpp/35a_1294A_CollectingCoins.cpp
#include<bits/stdc++.h>

int main(){
    int t, a, b, c, n, sum;
    std::vector<std::string> ans;
    std::cin>>t;
    for(int i = 0; i < t; i++){
        std::cin>>a>>b>>c>>n;
        std::vector<int> array = {a,b,c};
        sort(array.begin(), array.end());
        sum = 0 + array[2] - array[0] + array[2] - array[1];
        if (sum > n){
            ans.push_back("NO");
        }
        else{
            n = n - sum;
            if(n %  3 == 0){
                ans.push_back("YES");
            }
            else{
                ans.push_back("NO");
            }
        }
    }
    for(auto i : ans){
        std::cout<<i<<"\n";
    }
    return 0;
}

File: python/35a_1294A_CollectingCoins.py
def main():
    ans = []
    t = int(input())
    for _ in range(t):
        a, b, c, n = input().split()
        a, b, c, n = int(a), int(b), int(c), int(n)
        array = [a, b, c]
        array.sort()
        Sum = 0 + (array[2] - array[0]) + (array[2] - array[1])
        if Sum > n:
            ans.append("NO")
        else:
            n = n - Sum
            if n % 3 == 0:
                ans.append("YES")
            else:
                ans.append("NO")
    for i in ans:
        print(i)

if __name__ == "__main__":
    main()